src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/TokenCertificateService.java

```diff
@@ -642,11 +700,16 @@ static boolean isCausedByCertNotFound(CodedException e) {
         return CERT_NOT_FOUND_FAULT_CODE.equals(e.getFaultCode());
     }
 
+    static boolean isCausedBySslAuthFailure(CodedException e) {
+        return SSL_AUTH_FAULT_CODE.equals(e.getFaultCode());
+    }
+
     static final String DUPLICATE_CERT_FAULT_CODE = signerFaultCode(X_CERT_EXISTS);
     static final String INCORRECT_CERT_FAULT_CODE = signerFaultCode(X_INCORRECT_CERTIFICATE);
     static final String CERT_WRONG_USAGE_FAULT_CODE = signerFaultCode(X_WRONG_CERT_USAGE);
     static final String CSR_NOT_FOUND_FAULT_CODE = signerFaultCode(X_CSR_NOT_FOUND);
     static final String CERT_NOT_FOUND_FAULT_CODE = signerFaultCode(X_CERT_NOT_FOUND);
+    static final String SSL_AUTH_FAULT_CODE = clientProxyFaultCode(X_CERT_NOT_FOUND);
```

Comment:
This is a duplicate of CERT_NOT_FOUND_FAULT_CODE, on purpose or not? Anyways probably deprecated if unregisterAuthCertAndMarkForDeletion exception handling is refactored.

Reply:
Sorry, no. Did not see the clientProxy vs signer part initially.